base/math.jl

```diff
@@ -349,8 +349,45 @@ julia> ldexp(5., 2)
 20.0
 ```
 """
-ldexp(x::Float64,e::Integer) = ccall((:scalbn,libm),  Float64, (Float64,Int32), x, Int32(e))
-ldexp(x::Float32,e::Integer) = ccall((:scalbnf,libm), Float32, (Float32,Int32), x, Int32(e))
+function ldexp{T<:AbstractFloat}(x::T, q::Integer)
+    n = Int(q)
```

Comment:
The reason I have this is if  `q::Int128` then things are type unstable with `k + q` . It's not a problem for Int32 or Int64